packages/maskbook/src/web3/UI/EthereumGasButton.tsx

```diff
@@ -0,0 +1,371 @@
+                            <Typography variant="body2" color="textSecondary">
+                                &#8776; ${new BigNumber(usd || '0').multipliedBy(eth).toFixed(6)}
+                            </Typography>
+                        </div>
+                    </div>
+                </div>
+            </ButtonBase>
+        </Grid>
+    )
+}
+
+const useGasPricesLitsStyles = makeStyles((theme) =>
+    createStyles({
+        root: {
+            width: '100%',
+            display: 'flex',
+            flexDirection: 'row',
+            alignItems: 'center',
+            justifyContent: 'center',
+        },
+    }),
+)
+
+interface GasPricesListProps {
+    gasPrices: GasPrice[]
+    selectedIndex: number
+    usd?: string
+    onChange?: (index: number, gasPrice: GasPrice) => void
+}
+
+export function GasPricesList(props: GasPricesListProps) {
+    const classes = useGasPricesLitsStyles()
+    const { gasPrices = [], onChange, selectedIndex = 0, usd } = props
+    const [selected, setSelected] = useState(selectedIndex)
+
+    const handleOnChange = useCallback(
+        (index: number, gasPrice: GasPrice) => {
+            if (selected !== index) {
+                setSelected(index)
+            }
+            onChange?.(index, gasPrice)
+        },
+        [selected, onChange],
+    )
+    return (
+        <div className={classes.root}>
+            {gasPrices.map((item, index) => (
+                <Fragment key={index}>
+                    <GasPriceItem
+                        usd={usd}
+                        key={index}
+                        active={selected === index}
+                        gasPrice={item}
+                        onChange={(gasPrice: GasPrice) => handleOnChange(index, gasPrice)}
+                    />
+                </Fragment>
+            ))}
+        </div>
+    )
+}
+
+const useDialogStyles = makeStyles((theme) =>
+    createStyles({
+        content: {
+            padding: theme.spacing(2, 4),
+        },
+        hit: {
+            padding: theme.spacing(1),
+        },
+        dialogAction: {
+            display: 'flex',
+            justifyContent: 'center',
+            padding: theme.spacing(2, 4),
+        },
+    }),
+)
+
+interface EthereumGasDialogProps {
+    gasPrices?: GasPrice[]
+    open: boolean
+    usd?: string
+    onSubmit?: (gasPrice?: GasPrice) => void
+    onClose: () => void
+}
+
+function EthereumGasDialog(props: EthereumGasDialogProps) {
+    const { t } = useI18N()
+    const classes = useDialogStyles()
+    const { gasPrices = [], onClose, onSubmit, usd } = props
+    const [gasPrice, setGasPrice] = useState<GasPrice | null>(gasPrices && gasPrices.length > 0 ? gasPrices[0] : null)
+    const [selected, setSeleted] = useState(0)
+
+    const onChange = (index: number, gasPrice: GasPrice) => {
+        setGasPrice(gasPrice)
+        setSeleted(index)
+    }
+    const handleClick = useCallback(() => {
+        onSubmit?.(gasPrice!)
```

Comment:
It looks like the `non-null assertion operator` is not necessary.

Reply:
if gas price is custom , it  necessary

Reply:
But the `onSubmit` can accept a nullish value according to its signature.